rts/System/Platform/Linux/InterprocessRecursiveMutex.cpp

```diff
@@ -0,0 +1,119 @@
+#include "InterprocessRecursiveMutex.h"
+
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <system_error>
+
+#include "System/Platform/Misc.h"
+#include "System/Misc/SpringTime.h"
+
+InterprocessRecursiveMutex::InterprocessRecursiveMutex(const char* name_) noexcept(false)
+	: name(name_)
+	, shmFd(-1)
+	, shData(nullptr)
+{
+	shmFd = shm_open(name.c_str(), O_CREAT | O_RDWR, 0660);
+	if (shmFd == -1) {
+		const auto errNum = errno;
+		throw std::system_error(errNum, std::generic_category(), Platform::GetLastErrorAsString(errNum));
+	}
+
+	if (ftruncate(shmFd, sizeof(shared_data)) == -1) {
+		const auto errNum = errno;
+		throw std::system_error(errNum, std::generic_category(), Platform::GetLastErrorAsString(errNum));
+	}
+
+	// Map the shared memory segment
+	shData = reinterpret_cast<shared_data*>(mmap(nullptr, sizeof(shared_data), PROT_READ | PROT_WRITE, MAP_SHARED, shmFd, 0));
+	if (shData == MAP_FAILED) {
+		const auto errNum = errno;
+		throw std::system_error(errNum, std::generic_category(), Platform::GetLastErrorAsString(errNum));
```

Comment:
Looks like this error is copypasted everywhere, consider something like
```cpp
const auto throw_errno = []() {
  const auto errNum = errno;
  throw std::system_error(errNum, std::generic_category(), Platform::GetLastErrorAsString(errNum));
};
```
and then just
```
if (this_failed)
  throw_errno();

if (that_failed)
  throw_errno();

if (yet_another_hting_failed)
  throw_errno();
```